python/meterpreter/meterpreter.py

```diff
@@ -785,12 +786,18 @@ def _read1(self):
             return bytes()
 
     def run(self):
-        byte = self._read1()
-        while len(byte):
-            self.data_lock.acquire()
-            self.data += byte
-            self.data_lock.release()
+        try:
             byte = self._read1()
+            while len(byte) > 0:
+                self.data_lock.acquire()
+                self.data += byte
+                self.data_lock.release()
+                byte = self._read1()
+        finally:
+            self._is_reading = False
+
+    def is_reading(self):
```

Comment:
I didn't want to call this `is_alive`, as would shadow `threading.Thread#is_alive` which seems semantically disparate